src/test/test-seccomp.c

```diff
@@ -854,8 +854,7 @@ static void test_load_syscall_filter_set_raw(void) {
                 assert_se(s = hashmap_new(NULL));
 #if defined __NR_poll && __NR_poll >= 0
                 assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_poll + 1), INT_TO_PTR(EILSEQ)) >= 0);
-#elif defined __NR_ppoll
-                assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_ppoll + 1), INT_TO_PTR(-1)) >= 0);
+#elif defined __NR_ppoll && __NR_ppoll >= 0
                 assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_ppoll + 1), INT_TO_PTR(EILSEQ)) >= 0);
 #endif
```

Comment:
hmm, this does not look right to me. At least I don't understand how this is supposed to work. I mean, if riscv32 has neither poll() nor ppoll(), and hence this filter is empty, how come the check below that checks that poll() fails with EILSEQ doesn't fail?

there's something strange afoot here. I don#t get it...

I think the `#elif` here should be a regular `#if`, i.e. we should block all flavours of poll(), not just the first one we find defined. Moreover, given that riscv32 appears to have ppoll_time32 (or how was that called?) we should block that here, too.

Reply:
@yuwata: what about`ppoll_time32` here?

Reply:
Oh, it's mapped to the same syscall number, based on config options? Anyway, I think it's fine to ignore this issue for now. Let's get this merged.

Reply:
@keszybz ppoll_time32?? instead of ppoll_time64?

@kraj ppoll_time64 is already added in the revised commit. Thanks.